docs/src/charm/howto/openstack.md

```diff
@@ -0,0 +1,104 @@
+# Integrating with OpenStack
+
+This guide explains how to integrate {{product}} with the OpenStack cloud
+platform.
+
+## Prerequisites
+
+To follow this guide, you will need:
+
+- An [OpenStack][openstack] cloud environment.
+- Juju set-up as the deployment tool for Openstack. Please refer to
+  [Juju and Openstack][juju-openstack] documentation for more information.
+- An OpenStack [project][project] with the necessary permissions to deploy
+  {{product}}.
+- A juju [controller][controller] with [access][credentials] to the OpenStack 
+  cloud environment.
+- A juju [model][model] for deploying {{product}} on OpenStack.
+- Configured [proxy configuration][proxy] in constrained environments.
+
+## Deploying {{product}} on OpenStack
+
+
+To deploy the {{product}} [bundle][bundle] on OpenStack you need an overlay
+bundle which serves as an extension of the core bundle. The overlay bundle
+contains the necessary configuration to deploy {{product}} on OpenStack.
+
+Refer to the base overlay [openstack-lb-overlay.yaml][openstack-overlay] and
+modify it as needed.
+
+### OpenStack overlay configurations:
+
+Run `openstack project list` to retrieve your project id and include the
+[project id][project] in the overlay template:
+
+```yaml
+applications:
+  openstack-integrator:
+    options:
+      project-id: <my-openstack-project-id>
+```
+
+
+Adjust [easyrsa][easyrsa] to avoid creating an LXD machine:
+
+```yaml
+applications:
+  easyrsa:
+    to:
+      - 0
+```
+
+The modified overlay template should look like this:
+
+```yaml
+applications:
+  kubeapi-load-balancer: null
+  kubernetes-control-plane:
+    options:
+      allow-privileged: "true"
+  openstack-integrator:
+    charm: openstack-integrator
+    num_units: 1
+    trust: true
+    options:
+      project-id: <my-openstack-project-id>
+  openstack-cloud-controller:
+    charm: openstack-cloud-controller
+  cinder-csi:
+    charm: cinder-csi
+  easyrsa: 
+    to:   
+      - 0 
+relations:
+  - [openstack-cloud-controller:certificates,            easyrsa:client]
+  - [openstack-cloud-controller:kube-control,            kubernetes-control-plane:kube-control]
```

Comment:
I think we should change these. For Canonical K8s, it's `k8s` and `k8s-worker`.
Also I'm now thinking that we might need to include an Openstack overlay (or overlay template) somewhere in the `k8s-bundles` repo which is sort of customized for our Canonical K8s charm. 
Overall, I think this overlay needs significant changes to work with Canonical K8s (in the K8s Openstack docs I only tested against `kubernetes-core` bundle).

Reply:
Agreed we need the k8s template in the k8s-bundles repo and tested before these docs can get published.

Reply:
@HomayoonAlimohammadi Are you working on verifying the Canonical k8s overlay template already?

Reply:
No I will create a card for that, but we don't have that for now.